ogr/ogrsf_frmts/arrow_common/ograrrowlayer.hpp

```diff
@@ -0,0 +1,2389 @@
+            {
+                const auto castArray = static_cast<const arrow::Int64Array*>(array);
+                poFeature->SetField(i, static_cast<GIntBig>(castArray->Value(nIdxInBatch)));
+                break;
+            }
+            case arrow::Type::HALF_FLOAT:
+            {
+                const auto castArray = static_cast<const arrow::HalfFloatArray*>(array);
+                poFeature->SetField(i, castArray->Value(nIdxInBatch));
+                break;
+            }
+            case arrow::Type::FLOAT:
+            {
+                const auto castArray = static_cast<const arrow::FloatArray*>(array);
+                poFeature->SetField(i, castArray->Value(nIdxInBatch));
+                break;
+            }
+            case arrow::Type::DOUBLE:
+            {
+                const auto castArray = static_cast<const arrow::DoubleArray*>(array);
+                poFeature->SetField(i, castArray->Value(nIdxInBatch));
+                break;
+            }
+            case arrow::Type::STRING:
+            {
+                const auto castArray = static_cast<const arrow::StringArray*>(array);
+                poFeature->SetField(i, castArray->GetString(nIdxInBatch).c_str());
+                break;
+            }
+            case arrow::Type::BINARY:
+            {
+                const auto castArray = static_cast<const arrow::BinaryArray*>(array);
+                int out_length = 0;
+                const uint8_t* data = castArray->GetValue(nIdxInBatch, &out_length);
+                poFeature->SetField(i, out_length, data);
+                break;
+            }
+            case arrow::Type::FIXED_SIZE_BINARY:
+            {
+                const auto castArray = static_cast<const arrow::FixedSizeBinaryArray*>(array);
+                const uint8_t* data = castArray->GetValue(nIdxInBatch);
+                poFeature->SetField(i, castArray->byte_width(), data);
+                break;
+            }
+            case arrow::Type::DATE32:
+            {
+                // number of days since Epoch
+                const auto castArray = static_cast<const arrow::Date32Array*>(array);
+                int64_t timestamp = static_cast<int64_t>(castArray->Value(nIdxInBatch)) * 3600 * 24;
+                struct tm dt;
+                CPLUnixTimeToYMDHMS(timestamp, &dt);
+                poFeature->SetField(i, dt.tm_year + 1900, dt.tm_mon + 1, dt.tm_mday,
+                                    0,0,0);
+                break;
+            }
+            case arrow::Type::DATE64:
+            {
+                // number of milliseconds since Epoch
+                const auto castArray = static_cast<const arrow::Date64Array*>(array);
+                int64_t timestamp = static_cast<int64_t>(castArray->Value(nIdxInBatch)) / 1000;
+                struct tm dt;
+                CPLUnixTimeToYMDHMS(timestamp, &dt);
+                poFeature->SetField(i, dt.tm_year + 1900, dt.tm_mon + 1, dt.tm_mday,
```

Comment:
Not sure to follow by you mean by "in the field handling code we are using integer 64 type for DATE64 arrow types": in OGRArrowLayer::MapArrowTypeToOGR() ?  You're perhaps confusing with TIME64, which is indeed mapped to a Integer64

DATE64 arrow types are transported as int64 (they are a logical type over a int64 which is a number of milliseconds since 1970-01-01 00:00:00), but at the OGR side we want to use a decoded view as a OFTDateTime

Reply:
You're right, sorry. I mixed up the TIME64/DATE64 handling.